client/AccountsClient.ts

```diff
@@ -62,95 +58,130 @@ export default class AccountsClient {
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.CHECKOUT, [request]);
     }
 
-    public logout(request: SimpleRequest, requestBehavior = this._defaultBehavior): Promise<SimpleResult> {
+    public logout<B extends BehaviorType = DB>(
+        request: SimpleRequest,
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.LOGOUT, [request]);
     }
 
-    public export(
+    public export<B extends BehaviorType = DB>(
         request: ExportRequest,
-        requestBehavior = this._defaultBehavior,
-    ): Promise<ExportResult> {
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.EXPORT, [request]);
     }
 
-    public changePassword(
+    public changePassword<B extends BehaviorType = DB>(
         request: SimpleRequest,
-        requestBehavior = this._defaultBehavior,
-    ): Promise<SimpleResult> {
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.CHANGE_PASSWORD, [request]);
     }
 
-    public addAddress(request: SimpleRequest, requestBehavior = this._defaultBehavior): Promise<Address> {
+    public addAddress<B extends BehaviorType = DB>(
+        request: SimpleRequest,
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.ADD_ADDRESS, [request]);
     }
 
-    public rename(request: RenameRequest, requestBehavior = this._defaultBehavior): Promise<Account> {
+    public rename<B extends BehaviorType = DB>(
+        request: RenameRequest,
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.RENAME, [request]);
     }
 
-    public signMessage(
+    public signMessage<B extends BehaviorType = DB>(
         request: SignMessageRequest,
-        requestBehavior = this._defaultBehavior,
-    ): Promise<SignedMessage> {
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.SIGN_MESSAGE, [request]);
     }
 
-    public migrate(requestBehavior = this._defaultBehavior): Promise<Account[]> {
+    public migrate<B extends BehaviorType = DB>(
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
         return this._request(requestBehavior, RequestType.MIGRATE, [{ appName: 'Accounts Client' }]);
     }
 
     /**
      * Only accessible in iframe from Nimiq domains.
      */
-    public list(requestBehavior = this._iframeBehavior): Promise<Account[]> {
+    public list<B extends BehaviorType = DB>(
+        requestBehavior: RequestBehavior<B> = this._iframeBehavior as any,
```

Comment:
Shouldn't this ideally be restricted to `IFrameRequestBehavior`?

Reply:
Could be. I guess it's kind of a backup option for Apple devices...